Program.cs

```diff
@@ -1,36 +1,59 @@
 ﻿using System;
 using Microsoft.VisualBasic.FileIO;
 using System.Collections.Generic;
+using NLog;
+using NLog.Config;
+using NLog.Targets;
+using System.Globalization;
 
 namespace SupportBank
 {
     class Program
     {
-        static Dictionary<string, Account> ReadFile() {
+        private static readonly ILogger logger = LogManager.GetCurrentClassLogger();
 
-            var accounts = new Dictionary<string, Account>();
-
-            var parser = new TextFieldParser(@"Transactions2014.csv")
+        static Dictionary<string, Account> ReadFile(string filename)
+        {
+            logger.Info($"Initialising parser for: { filename }.");
+            var parser = new TextFieldParser(filename)
             {
                 TextFieldType = FieldType.Delimited
             };
             parser.SetDelimiters(",");
+            logger.Info("Parser initialized, parsing file.");
 
             // Skip first line
             if (!parser.EndOfData)
             {
                 parser.ReadLine();
             }
 
+            var accounts = new Dictionary<string, Account>();
+            var line = 0;
+
             while (!parser.EndOfData)
             {
+                line++;
                 var fields = parser.ReadFields();
 
                 var date = fields[0];
                 var from = fields[1];
                 var to = fields[2];
                 var narrative = fields[3];
-                var amount = decimal.Parse(fields[4]);
+
+                if (!DateTime.TryParseExact(fields[0], "dd/MM/yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime dateTime))
+                {
+                    string message = $"Skipped line { line } due to invalid date: { fields[0] }.";
```

Comment:
Minor but string interpolation is one of the few cases where we generally wouldn't add whitespace after an opening curly bracket and before a closing curly bracket 🙂